src/graphql/schema/uuid/abstract-uuid/resolvers.ts

```
@@ -30,11 +30,31 @@ export const resolvers: UuidResolvers = {
   },
   Query: {
     async uuid(_parent, payload, { dataSources }) {
-      const id = payload.alias
-        ? (await dataSources.serlo.getAlias(payload.alias)).id
-        : (payload.id as number)
+      if (payload.alias) {
+        const cleanPath = decodePath(payload.alias.path)
+        const match = /^\/(\d+)$/.exec(cleanPath)
+        if (match) {
+          const id = parseInt(match[1], 10)
+          return dataSources.serlo.getUuid<UuidPayload>({ id })
+        }
 
-      return dataSources.serlo.getUuid<UuidPayload>({ id })
+        if (cleanPath.startsWith('/user/profile/')) {
+          const match = /^\/user\/profile\/(\d+)$/.exec(cleanPath)
+          if (match) {
+            const id = parseInt(match[1], 10)
+            const uuid = await dataSources.serlo.getUuid<UuidPayload>({ id })
+            return uuid && uuid.__typename === 'User' ? uuid : null
+          }
+        }
+        const alias = await dataSources.serlo.getAlias(payload.alias)
+        return alias
+          ? dataSources.serlo.getUuid<UuidPayload>({ id: alias.id })
+          : null
+      } else if (payload.id) {
+        return dataSources.serlo.getUuid<UuidPayload>({ id: payload.id })
+      } else {
+        return null
```

Or shall we return an error here? (before we would have done a request to `/api/uuid/undefined` at serlo.org)